### PR TITLE
Dev into Master

### PR DESCRIPTION
**Pull request by karimkefi**
Completed dev and code reviewed. This is for Mike and Charlie review.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1 +1,168 @@
-hello
+<!DOCTYPE html>
+<html lang="en">
+
```
Comment (mporam) [inline index.html]: why whitespace

File: index.html [hunk 2]
```diff
@@ -1 +1,168 @@
...
+<head>
+    <meta charset="UTF-8">
+    <title>Karim Portfolio</title>
+    <link href="style.css" rel="stylesheet" type="text/css">
+	<link href="normalize.css" rel="stylesheet" type="text/css">
+</head>
+
```
Comment (mporam) [inline index.html]: ?

File: index.html [hunk 3]
```diff
@@ -1 +1,168 @@
-hello
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <title>Karim Portfolio</title>
+    <link href="style.css" rel="stylesheet" type="text/css">
+	<link href="normalize.css" rel="stylesheet" type="text/css">
```
Comment (mporam) [inline index.html:7]: why indented?

File: index.html [hunk 4]
```diff
@@ -1 +1,168 @@
...
+</head>
+
+<body>
+
```
Comment (mporam) [inline index.html]: ?

File: index.html [hunk 5]
```diff
@@ -1 +1,168 @@
...
+    <div class="headerImage1" id="#">
+
```
Comment (mporam) [inline index.html]: im going to stop commenting on whitespace now...but theres too much...

File: index.html [hunk 6]
```diff
@@ -1 +1,168 @@
...
+		<aside>
+			<ul class="sideNavList">
+				<li class="sideNavIcon">
+					<a href="#">
```
Comment (mporam) [inline index.html]: why does this link nowhere?

File: index.html [hunk 7]
```diff
@@ -1 +1,168 @@
...
+	<link href="normalize.css" rel="stylesheet" type="text/css">
+</head>
+
+<body>
+
+    <div class="headerImage1" id="#">
+
+		<aside>
+			<ul class="sideNavList">
+				<li class="sideNavIcon">
```
Comment (mporam) [inline index.html]: rather than having this class on all lis could you just say .sideNavList li in your css?

File: index.html [hunk 8]
```diff
@@ -1 +1,168 @@
...
+				</li>
+				<li class="sideNavIcon">
+					<a href="#contactLocation">
+						<img src="Img/iconsEmail.png" class="icon" alt="Email Icon">
+					</a>
+				</li>
+			</ul>
+		</aside>
+
+		<header>Karim Kefi Portfolio</header>
```
Comment (mporam) [inline index.html]: probably use a h1 here for google and semantics

File: index.html [hunk 9]
```diff
@@ -1 +1,168 @@
...
+<head>
+    <meta charset="UTF-8">
+    <title>Karim Portfolio</title>
+    <link href="style.css" rel="stylesheet" type="text/css">
+	<link href="normalize.css" rel="stylesheet" type="text/css">
+</head>
+
+<body>
+
+    <div class="headerImage1" id="#">
```
Comment (mporam) [inline index.html]: invalid id

File: index.html [hunk 10]
```diff
@@ -1 +1,168 @@
...
+
+		<aside>
```
Comment (mporam) [inline index.html]: is this an aside? just because its at the side, its still the nav bar, so probably use nav

File: index.html [hunk 11]
```diff
@@ -1 +1,168 @@
...
+					</a>
+				</li>
+			</ul>
+		</aside>
+
+		<header>Karim Kefi Portfolio</header>
+
+		<section class="topContainer">
+			<div class="codingLanguagesLarge codingLanguagesMedium">
+					<div class="codingIconLarge codingIconMedium">
```
Comment (mporam) [inline index.html]: too much indentation

File: style.css [hunk 12]
```diff
@@ -0,0 +1,314 @@
...
+    cursor: pointer;
+    background-color: #12aaff;
+    height: 100px;
+}
+
+.icon {
+    transform: translate(25%,25%);
+}
+
+/* xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx  */
```
Comment (mporam) [inline style.css]: helpful...

File: index.html [hunk 13]
```diff
@@ -1 +1,168 @@
...
+						<img src="Img/iconsEmail.png" class="icon" alt="Email Icon">
+					</a>
+				</li>
+			</ul>
+		</aside>
+
+		<header>Karim Kefi Portfolio</header>
+
+		<section class="topContainer">
+			<div class="codingLanguagesLarge codingLanguagesMedium">
```
Comment (mporam) [inline index.html]: why 2 classes? u can just use 1 class with no size on it and change it in the media query

File: index.html (hunk 11 above)
Comment (mporam) [inline index.html]: same as other principal, dont need both classes

File: index.html [hunk 14]
```diff
@@ -1 +1,168 @@
...
+				As the challenges get better (and the old ones get embarassing), this
+				portfolio section will change!
+			</p>
+		</section>
+	</div>
+
+	<footer class="footer" id="contactLocation">
+
+		<ul class="bottomContainer">
+            <div class="footerBoxLarge footerBoxMedium footerBoxSmall">
```
Comment (mporam) [inline index.html]: only `<li>` tags allowed as direct childs of `<ul>`
also, fix ALL the indentation. Option + CMD + L will do this for you.

File: style.css [hunk 15]
```diff
@@ -0,0 +1,314 @@
...
+    text-align: center;
+}
+
+.breatherContainerLarge p,
+.breatherContainerLarge h1 {
+    margin: 20px;
+    width: 90%;
+}
+
+.bottomContainer {
```
Comment (mporam) [inline style.css]: alot of these containers have similar styles. can you put the common styles into a generic class then the different styles into modifier classes?

File: style.css [hunk 16]
```diff
@@ -0,0 +1,314 @@
...
+    position: relative;
+    margin: 0 auto;
+    height: 250px;
+    width: 85%;
+    overflow: auto;
+    text-align: center;
+}
+
+/* xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx  */
```
Comment (mporam) [inline style.css]: whats this?

File: style.css [hunk 17]
```diff
@@ -0,0 +1,314 @@
...
+.headerBox {
+    position: fixed;
+    height: 140px;
+    width:100%;
+}
+
+header {
+    text-align: center;
+    top: 20px;
+	height: 80px;
```
Comment (mporam) [inline style.css:67]: weird indentation

File: style.css [hunk 18]
```diff
@@ -0,0 +1,314 @@
...
+	background-color: RGBA(64,64,64,0);
+	font-size: 3rem;
+	transition-timing-function: ease;
+	transition: width 1s;
+    cursor: pointer;
+}
+
+header:hover {
+    background-color: #1091DA;
+	width: 100%;
```
Comment (mporam) [inline style.css:79]: indentation

File: style.css [hunk 19]
```diff
@@ -0,0 +1,314 @@
...
+	cursor: pointer;
+}
+
+/* xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx  */
```
Comment (mporam) [inline style.css]: again, what is this?

File: style.css [hunk 20]
```diff
@@ -0,0 +1,314 @@
...
+	width: 600px;
+	background-color: RGBA(64,64,64,0);
+	font-size: 3rem;
+	transition-timing-function: ease;
+	transition: width 1s;
+    cursor: pointer;
+}
+
+header:hover {
+    background-color: #1091DA;
```
Comment (mporam) [inline style.css]: either all lowercase or all uppercase hex codes please

File: style.css [hunk 21]
```diff
@@ -0,0 +1,314 @@
...
+}
+
+/* xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx  */
+
+aside {
+    position: fixed;
+    right: 0px;
+    width: 90px;
+    height:100%;
+    display: inline-block;
```
Comment (mporam) [inline style.css]: this does nothing when its position fixed

File: style.css [hunk 22]
```diff
@@ -0,0 +1,314 @@
...
+    background-color: #1091da;
+    z-index: 5;
+}
+
+.sideNavList {
+    padding: 0px;
```
Comment (mporam) [inline style.css]: px is not needed

File: style.css [hunk 23]
```diff
@@ -0,0 +1,314 @@
...
+    margin-top: 20px;
+    float: left;
+}
+
+.sideNavIcon {
+    width: 50px;
+    height: 50px;
+    border: 1px solid black;
```
Comment (mporam) [inline style.css]: hex

File: style.css [hunk 24]
```diff
@@ -0,0 +1,314 @@
...
+    top: 50px;
+    display: block;
+    width: 100%;
+    height: 120px;
+    overflow: auto;
+}
+
+.codingIconLarge {
+    float: left;
+    display:inline-block;
```
Comment (mporam) [inline style.css]: this does nothing on a floated element, float makes it display block

File: style.css [hunk 25]
```diff
@@ -0,0 +1,314 @@
...
+    width: 20%;
+    height:100px;
```
Comment (mporam) [inline style.css]: space after :

File: style.css [hunk 26]
```diff
@@ -0,0 +1,314 @@
...
+}
+
+/* xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx  */
+
+.homepageText {
+    color: #FFFFFF;
```
Comment (mporam) [inline style.css]: lowercase

File: style.css [hunk 27]
```diff
@@ -0,0 +1,314 @@
...
+    background-color: #0f5285;
+    font-size: 1.2rem;
+}
+
+/* xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx  */
+
+.headerImage1 {
```
Comment (mporam) [inline style.css]: the only difference between these classes is the image, make 1 generic class with the generic styles and then 3 modifier classes with the images on

File: style.css [hunk 28]
```diff
@@ -0,0 +1,314 @@
...
+    background-size: cover;
+    min-height: 70vh;
+}
+
+/* xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx  */
+
+.articleBoxLarge {
+    overflow: auto;
+    height: 200px;
+    margin:0 auto;
```
Comment (mporam) [inline style.css]: space after :

File: style.css [hunk 29]
```diff
@@ -0,0 +1,314 @@
...
+.articleBoxLarge img {
+    height: 160px;
+    width: 320px;
+    float: left;
+    margin: 15px;
+    padding: 0 15px;
+}
+
+.articleBoxLarge p {
+    display: inline-block;
```
Comment (mporam) [inline style.css]: does nothing

File: style.css [hunk 30]
```diff
@@ -0,0 +1,314 @@
...
+}
+
+/* xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx  */
+
+.footer {
+    position: relative;
+    width: 100%;
+    height: 250px;
+    margin: 0 auto;
+    color: #FFFFFF;
```
Comment (mporam) [inline style.css:198]: lowercase

File: style.css [hunk 31]
```diff
@@ -0,0 +1,314 @@
...
+
+.footerBoxLarge {
+    height: 160px;
+    margin: 15px;
+    width: 30%;
+    float: left;
+}
+
+.footerIcon {
+    display: inline-block;
```
Comment (mporam) [inline style.css]: does nothing

File: style.css [hunk 32]
```diff
@@ -0,0 +1,314 @@
...
+    margin: 0 auto;
+    float: left;
+    width: 30%x;
```
Comment (mporam) [inline style.css]: x?

**Review by mporam — CHANGES_REQUESTED** [review]

**Comment by mporam** [comment]
doesnt work on small mobile size.
Please centre content and make bigger on mobile, is very hard to read font size.

File: index.html (hunk 3 above)
Comment (karimkefi) [reply index.html:7]: weird, not indented on PHPStorm

File: index.html (hunk 5 above)
Comment (karimkefi) [reply index.html]: removed all

File: index.html (hunk 6 above)
Comment (karimkefi) [reply index.html]: # defaults to the top of the page

Comment (karimkefi) [reply index.html]: the hash on its own takes you to the top of the page. I have changed this to make the anchor clear (now called "home")

File: index.html (hunk 9 above)
Comment (karimkefi) [reply index.html]: the hash on its own takes you to the top of the page. I have changed this to make the anchor clear (now called "home")

File: index.html (hunk 8 above)
Comment (karimkefi) [reply index.html]: I have left as "header" as this has CSS on the whole element sliding left to right

File: index.html (hunk 11 above)
Comment (karimkefi) [reply index.html]: done

File: style.css (hunk 12 above)
Comment (karimkefi) [reply style.css]: I have made the comments clearer

File: index.html (hunk 10 above)
Comment (karimkefi) [reply index.html]: changed to NAV

File: style.css (hunk 17 above)
Comment (karimkefi) [reply style.css:67]: weird as its ok in PHP Storm (i.e. no extra indentation)

File: style.css (hunk 18 above)
Comment (karimkefi) [reply style.css:79]: weird as its ok in PHP Storm (i.e. no extra indentation)

File: style.css (hunk 19 above)
Comment (karimkefi) [reply style.css]: updated all comment lines

File: style.css (hunk 22 above)
Comment (karimkefi) [reply style.css]: changed

File: style.css (hunk 23 above)
Comment (karimkefi) [reply style.css]: updated all

File: style.css (hunk 24 above)
Comment (karimkefi) [reply style.css]: removed

File: style.css (hunk 28 above)
Comment (karimkefi) [reply style.css]: done

File: style.css (hunk 32 above)
Comment (karimkefi) [reply style.css]: corrected

File: style.css (hunk 30 above)
Comment (karimkefi) [reply style.css:198]: changed all consistently to capitals

File: index.html (hunk 1 above)
Comment (karimkefi) [reply index.html]: removed

File: index.html (hunk 4 above)
Comment (karimkefi) [reply index.html]: removed blank rows

File: index.html (hunk 7 above)
Comment (karimkefi) [reply index.html]: thanks, updated. much cleaner!

File: index.html (hunk 2 above)
Comment (karimkefi) [reply index.html]: removed line spacing in html

File: index.html (hunk 13 above)
Comment (karimkefi) [reply index.html]: Changed for all medial queries classes where I had Large Medium and Small

File: index.html (hunk 11 above)
Comment (karimkefi) [reply index.html]: Changed for all medial queries classes where I had Large Medium and Small

File: style.css (hunk 16 above)
Comment (karimkefi) [reply style.css]: updated all the comment lines to make them clearer

File: style.css (hunk 20 above)
Comment (karimkefi) [reply style.css]: updated all to upper case

File: style.css (hunk 21 above)
Comment (karimkefi) [reply style.css]: removed

File: style.css (hunk 29 above)
Comment (karimkefi) [reply style.css]: removed

File: style.css (hunk 31 above)
Comment (karimkefi) [reply style.css]: removed

File: style.css (hunk 25 above)
Comment (karimkefi) [reply style.css]: now consistent

File: style.css (hunk 26 above)
Comment (karimkefi) [reply style.css]: made all consistently uppercase

File: style.css (hunk 27 above)
Comment (karimkefi) [reply style.css]: thanks, changed and much cleaner

File: style.css (hunk 15 above)
Comment (karimkefi) [reply style.css]: updated these

File: index.html (hunk 14 above)
Comment (karimkefi) [reply index.html]: fixed this illegal move!

File: index.html [hunk 33]
```diff
@@ -1 +1,147 @@
...
+        </section>
+	</div>
+	<div>
+		<section class="breatherContainer breatherText">
+			<h1>SOME STUFF I HAVE BUILT</h1>
+			<p>Started at Mayden Academy in Feb 2018. During my first 2 weeks
+				we were set a challenges which are shown above.
+				As the challenges get better (and the old ones get embarrassing), this
+				portfolio section will change!
+			</p>
```
Comment (o-tahboub) [inline index.html]: P tag placed differently from other places in the rest of the code.

File: style.css [hunk 34]
```diff
@@ -0,0 +1,306 @@
...
+    }
+
+    .breatherContainer {
+        height: 175px;
+        }
+
+    .breatherContainer p,
+    .breatherContainer h1 {
+        margin: 20px;
+        /*width: 650px;*/
```
Comment (o-tahboub) [inline style.css]: remove comments from live code.

**Review by o-tahboub — COMMENTED** [review]
Sent you a couple of comments, other than that, its fine.

File: style.css (hunk 34 above)
Comment (karimkefi) [reply style.css]: removed

File: index.html (hunk 33 above)
Comment (karimkefi) [reply index.html]: updated in 2 places

**Review by o-tahboub — APPROVED** [review]
Issues highlighted in code review were fixed and committed.

**Comment by charliecog** [comment]
Looks good - approved!

**Pull request merged**